JDL code generation error on Jhipster5@Beta

<!--
- Please follow the issue template below for bug reports and feature requests.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**
Trying to generate project using jhjdl command using sample jdl file https://start.jhipster.tech/jdl-studio/, getting below error .
<!-- Explain the bug or feature request, if an error is being thrown a stack trace helps -->
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.7.0:compile (default-compile) on project jhtest: Compilation failure
[ERROR] /Volumes/Data/projects/jhdemo2/jhtest/src/main/java/com/gzs/audit/web/rest/JobResource.java:[112,46] no suitable method found for toDto(java.util.List<com.gzs.audit.service.dto.JobDTO>)
[ERROR]     method com.gzs.audit.service.mapper.EntityMapper.toDto(com.gzs.audit.domain.Job) is not applicable
[ERROR]       (argument mismatch; java.util.List<com.gzs.audit.service.dto.JobDTO> cannot be converted to com.gzs.audit.domain.Job)
[ERROR]     method com.gzs.audit.service.mapper.EntityMapper.toDto(java.util.List<com.gzs.audit.domain.Job>) is not applicable
[ERROR]       (argument mismatch; java.util.List<com.gzs.audit.service.dto.JobDTO> cannot be converted to java.util.List<com.gzs.audit.domain.Job>)
[ERROR]     method com.gzs.audit.service.mapper.JobMapper.toDto(com.gzs.audit.domain.Job) is not applicable
[ERROR]       (argument mismatch; java.util.List<com.gzs.audit.service.dto.JobDTO> cannot be converted to com.gzs.audit.domain.Job)##### **Motivation for or Use Case**
`
<!-- Explain why this is a bug or a new feature for you -->

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- For bug reports, if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

<!--
Which version of JHipster are you using, is it a regression?
-->
5.0@Beta
##### **JHipster configuration**

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->
https://start.jhipster.tech/jdl-studio/
##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->
MAC High Sierra
- [ ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs
+++ b/generators/entity-server/templates/src/main/java/package/common/get_all_template.ejs
@@ -72,7 +72,7 @@
         HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/<%= entityApiUrl %>");
         <%_ } _%>
     <%_ } _%>
-        return new ResponseEntity<>(<% if (!viaService && dto === 'mapstruct' && fieldsContainOwnerManyToMany) { %><%= entityListToDtoListReference %>(page.getContent())<% } else { %>page.getContent()<% } %>, headers, HttpStatus.OK);
+        return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
     <%_ } _%>
     }
 <%_ } _%>

--- a/travis/samples/.jhipster/TestManyRelWithPaginationAndDTO.json
+++ b/travis/samples/.jhipster/TestManyRelWithPaginationAndDTO.json
@@ -1,0 +1,19 @@
+{
+    "fluentMethods": true,
+    "relationships": [
+        {
+            "relationshipName": "testMapstruct",
+            "otherEntityName": "testMapstruct",
+            "relationshipType": "many-to-many",
+            "otherEntityField": "id",
+            "ownerSide": true
+        }
+    ],
+    "fields": [],
+    "changelogDate": "20160208210534",
+    "dto": "mapstruct",
+    "service": "no",
+    "angularJSSuffix": "mySuffix",
+    "entityTableName": "test_many_many_pagination_dto",
+    "pagination": "pagination"
+}

--- a/travis/samples/.jhipster/TestMapstruct.json
+++ b/travis/samples/.jhipster/TestMapstruct.json
@@ -15,6 +15,13 @@
             "otherEntityRelationshipName": "testMapstruct"
         },
         {
+            "relationshipName": "testManyRelWithPaginationAndDTO",
+            "otherEntityName": "testManyRelWithPaginationAndDTO",
+            "relationshipType": "many-to-many",
+            "ownerSide": false,
+            "otherEntityRelationshipName": "testMapstruct"
+        },
+        {
             "relationshipName": "testOneToOne",
             "otherEntityName": "testOneToOne",
             "relationshipType": "one-to-one",

--- a/travis/scripts/01-generate-entities.sh
+++ b/travis/scripts/01-generate-entities.sh
@@ -106,6 +106,7 @@ elif [[ ( "$JHIPSTER" == *"mysql"* ) || ( "$JHIPSTER" == *"psql"* ) ]]; then
     moveEntity TestPagination
     moveEntity TestManyToOne
     moveEntity TestManyToMany
+    moveEntity TestManyRelWithPaginationAndDTO
     moveEntity TestOneToOne
     moveEntity TestCustomTableName
     moveEntity TestTwoRelationshipsSameEntity

